common/lazy-pages/src/lib.rs

```diff
@@ -105,27 +127,38 @@ pub fn update_lazy_pages_and_protect_again(
             new_mem_addr
         );
 
-        // Cannot panic, unless OS allocates wasm mem buffer
-        // in not aligned by native page addr.
-        gear_ri::set_wasm_mem_begin_addr(new_mem_addr);
-    }
+        Some(new_mem_addr)
+    } else {
+        None
+    };
 
     let new_mem_size = mem.size();
-    if new_mem_size > old_mem_size {
-        gear_ri::set_wasm_mem_size(new_mem_size.into());
+    let changed_size = match new_mem_size > old_mem_size {
+        true => Some(new_mem_size.raw()),
+        false => None,
+    };
```

Comment:
One can simplify it.
```suggestion
    let changed_size = (new_mem_size > old_mem_size).then_some(new_mem_size.raw());
```

Reply:
done